eglot-jl.el

```diff
@@ -71,10 +76,31 @@ Otherwise returns nil"
 (cl-defmethod project-roots ((project (head julia)))
   (list (cdr project)))
 
+(defun eglot-jl--sysimage-args ()
+  "Returns a list of command-line switches suitable for sysimage loading.
+Return nil if no system image has been found or if system images
+have been disabled altogether."
+  (when eglot-jl-enable-sysimage
+    (let ((julia-version (with-temp-buffer
+                           (call-process eglot-jl-julia-command nil t nil
+                                         "-e" "print(Base.VERSION)")
+                           (buffer-string))))
+      (car
+       (remove nil
+               (mapcar
+                (lambda (ext)
+                  (let ((path (expand-file-name
+                               (concat "eglot-jl." julia-version ext)
```

Comment:
This doesn't encode the version of LanguageServer and other packages into the filename. What happens when we release a new version of `eglot-jl` with an updated Manifest? I don't actually know how custom sysimages and project manifests interact. Apologies if this is covered somewhere, and I missed it.

Reply:
This is a very important issue, and probably a complex one, too. So we should take some time to discuss it.

A first point to consider is that all users installing and using `eglot-jl` from MELPA should have each version installed in a separate directory. For such users, there is therefore no risk of having an older sysimage (i.e. previously compiled for an older version of `eglot-jl` & its deps), mess up anything in a newer version of `eglot-jl`. What happens now for such users is that as soon as they update `eglot-jl`, no sysimage is found and they have to manually compile a new one (or wait for the 20s start-up time). If sysimage compilation was made automatic upon opt-in (as per my proposal above), then after each `eglot-jl` update a new sysimage compilation would be automatically spawned in he background (or the user would be prompted to before this happens).

For users who do not install `eglot-jl` via MELPA, but rather clone the repository somewhere and load the package from there, the situation is entirely different. Updating the git repository would be performed in place, and leave any existing sysimage there to be discovered and used even if it is out-of-date. What happens is that when the julia version gets updated, the sysimage becomes invalid. When the Manifest gets updated, any previously compiled sysimage remains valid, but out-of-date in the sense that it does not incorporate new developments in `LanguageServer` and `SymbolServer`.

For such users, we would need a mechanism to invalidate the sysimage when the Manifest changes. And the only option we have is to change the name so that no appropriate system image is found. I can think of a few ways to do it:
- get the versions of dependencies from `Pkg` (e.g. from `Pkg.status()`) and somehow include them in the sysimage file name. This makes it impossible to compute the sysimage name in elisp: it has to be done in julia, which adds to the aforementioned overhead;
- work directly from the Manifest file itself, for example computing a hash of its contents and incorporating that in the sysimage name. This makes it possible to compute the sysimage name in a consistent way both in elisp (using `secure-hash`) and in julia (using *e.g.* `SHA.sha1`)

Reply:
I think using the hash of the manifest or some truncated version of it is probably the cleanest solution here. I think that the borg package manager installs packages directly from their git repo, so it's probably a workflow that should be supported.